src/benchmarklib/benchmark_runner.hpp

```diff
@@ -33,7 +33,7 @@ class BenchmarkRunner {
                   std::unique_ptr<AbstractTableGenerator> table_generator, const nlohmann::json& context);
   ~BenchmarkRunner();
 
-  void run();
+  int run();
```

Comment:
I derived this from QApplication::run() (as it is similar to the BenchmarkRunner in that it takes over the application completely) returning the main() return code. Will doc, is that okay?

If main() computes its own return code, then we have 3 main()s doing this (file based, join order, tpch).

Reply:
Mmmh, not happy with it, but if you doc it appropriately, that should be fine.

Reply:
Assert at the end of "BenchmarkRunner::run()" is better?

Reply:
Why not fail right when the validation fails?

Reply:
Because I want to run the entire JOB and see which queries fail and which do not. I do not want the benchmark to crash on the first failed query, waiting for me to start it again with the remaining queries.

Reply:
> Assert at the end of "BenchmarkRunner::run()" is better?

Yes

Reply:
Looks weird if the app logs verification failure while the queries are running and then again, via the Assert, after all queries have finished, but have it as you like.

Reply:
oh, that sounds sooo passive agressive :D